From 9c4508c7ad7ff47aab001c4d7e996491c2a64d85 Mon Sep 17 00:00:00 2001
From: Oneric <oneric@oneric.stub>
Date: Thu, 22 Jan 2026 00:00:00 +0000
Subject: [PATCH] Don't run io_uring tests on kernels lacking support

Even if the native Alpine kernel doesn't support io_uring,
building postgres with uring support enabled may still be
beneficial for setups where Alpine containers are used
on a host with a non-Alpine kernel.
---
 src/test/modules/test_aio/t/001_aio.pl | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/test/modules/test_aio/t/001_aio.pl b/src/test/modules/test_aio/t/001_aio.pl
index 3f0453619e8..fc64d181480 100644
--- a/src/test/modules/test_aio/t/001_aio.pl
+++ b/src/test/modules/test_aio/t/001_aio.pl
@@ -125,7 +125,10 @@ sub have_io_uring
 	my $methods = $1;
 	note "supported io_method values are: $methods";
 
-	return ($methods =~ m/io_uring/) ? 1 : 0;
+	# ensure kernel also offers io_uring syscalls
+	my $res = system_log [qw(grep io_uring_setup /proc/kallsyms)];
+
+	return ($res == 0 and $methods =~ m/io_uring/) ? 1 : 0;
 }
 
 sub psql_like
-- 
2.47.3

